File: app/src/main/java/com/basitshop/IntroActivity.java
package com.basitshop;

import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.view.MotionEvent;
import android.view.View;
import android.view.Window;
import android.view.WindowInsets;
import android.view.WindowManager;
import android.widget.FrameLayout;

import androidx.appcompat.app.AppCompatActivity;

public class IntroActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Fullscreen (hide status + nav)
        Window window = getWindow();
        window.setFlags(
                WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS,
                WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS
        );

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            window.getInsetsController().hide(
                    WindowInsets.Type.statusBars() | WindowInsets.Type.navigationBars()
            );
        } else {
            window.getDecorView().setSystemUiVisibility(
                    View.SYSTEM_UI_FLAG_FULLSCREEN
                            | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
                            | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY
            );
        }

        setContentView(R.layout.activity_intro);

        FrameLayout swipeContainer = findViewById(R.id.swipe_container);
        View swipeThumb = findViewById(R.id.swipe_thumb);

        swipeThumb.setOnTouchListener(new View.OnTouchListener() {
            float downX;
            boolean done = false;

            @Override
            public boolean onTouch(View v, MotionEvent e) {
                switch (e.getAction()) {
                    case MotionEvent.ACTION_DOWN:
                        downX = e.getRawX();
                        return true;

                    case MotionEvent.ACTION_MOVE:
                        float dx = e.getRawX() - downX;
                        float max = swipeContainer.getWidth() - v.getWidth() - 8;
                        dx = Math.max(0, Math.min(dx, max));
                        v.setTranslationX(dx);

                        if (dx > swipeContainer.getWidth() * 0.7f && !done) {
                            done = true;
                            openLogin(v);
                        }
                        return true;

                    case MotionEvent.ACTION_UP:
                        if (!done) v.animate().translationX(0).setDuration(200).start();
                        return true;
                }
                return false;
            }

            void openLogin(View v) {
                v.animate().translationX(swipeContainer.getWidth())
                        .setDuration(250)
                        .withEndAction(() -> {
                            startActivity(new Intent(IntroActivity.this, LoginActivity.class));
                            finish();
                        }).start();
            }
        });
    }
}

File: app/src/main/java/com/basitshop/IntroActivity.kt
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.LinearLayout;
import androidx.appcompat.app.AppCompatActivity;

public class IntroActivity extends AppCompatActivity {

    ImageView shapeTop, shapeBottom, iconArrow;
    View containerMain;
    LinearLayout layoutButton;
    FrameLayout btnGlassAction;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_intro);

        // Hooks
        shapeTop = findViewById(R.id.shape_top);
        shapeBottom = findViewById(R.id.shape_bottom);
        containerMain = findViewById(R.id.container_main);
        layoutButton = findViewById(R.id.layout_button);
        btnGlassAction = findViewById(R.id.btn_glass_action);
        iconArrow = findViewById(R.id.icon_arrow);

        // Load Animations
        Animation animFromTop = AnimationUtils.loadAnimation(this, R.anim.from_top);
        Animation animFromBottom = AnimationUtils.loadAnimation(this, R.anim.from_bottom);
        Animation animLeftPop = AnimationUtils.loadAnimation(this, R.anim.left_pop_fade);
        Animation animButtonBounce = AnimationUtils.loadAnimation(this, R.anim.button_up_bounce);
        Animation animArrowShake = AnimationUtils.loadAnimation(this, R.anim.arrow_shake);

        // Set Animations
        shapeTop.startAnimation(animFromTop);
        shapeBottom.startAnimation(animFromBottom);
        containerMain.startAnimation(animLeftPop);
        
        // Button comes from bottom with bounce
        layoutButton.startAnimation(animButtonBounce);
        
        // Arrow keeps shaking/wiggling inside the button
        iconArrow.startAnimation(animArrowShake);

        // Click Listener to open Login Screen
        View.OnClickListener startAction = v -> {
            Intent intent = new Intent(IntroActivity.this, LoginActivity.class); // LoginActivity class name check kar lena
            startActivity(intent);
            // Optional: Finish this activity so user can't go back
            // finish();
        };

        btnGlassAction.setOnClickListener(startAction);
        layoutButton.setOnClickListener(startAction); // Text area click karne par bhi kaam karega
    }
}
